### PR TITLE
peer: always defer clean up of the block epoch client used for ping headers

### LINKED ISSUE #6143: queue: possible ConcurrentQueue message leak
Checked up on one of my nodes, saw higher than usual memory allocated, which prompted me to dive in and take a few profiles. I think we might be forgetting to full drain the items in a concurrent queue when the client at the other end hangs up, or gets garbage collected itself. This causes the objects to just build up in memory, which starts to balloon the total space allocated. 

### Total Number of Inuse Objects
Screen Shot 2022-01-06 at 3 17 01 PM

### Total Inuse Space 
Screen Shot 2022-01-06 at 3 15 51 PM

### PR DESCRIPTION
**Pull request by Roasbeef**
In this commit, we fix an inadvertent memory leak by ensuring we always
use `defer` to clean up the allocated objects/memory we use to be
notified of new blocks to update what we send within the set of ping
headers.

A further optimization here would be using a single global block epoch
housed within the server, that all peer `pingHandler` goroutines use
directly.

Fixes #6143.

### CODE REVIEW

**Review by Crypt-iQ — APPROVED** [review]
yikes, can't believe i missed this. wonder if we could have a static check for this (all epochs get canceled)

**Comment by Roasbeef** [comment]
> yikes, can't believe i missed this. wonder if we could have a static check for this (all epochs get canceled)
Yeah that would be dope...

Spotted another thing I can do to make things more efficient as well, want to hold off on merging till that's in

**Review by carlaKC — APPROVED** [review]
LGTM, will re-review when additional improvement is pushed!

**Comment by Roasbeef** [comment]
Pushed up the other thing, it fixes a TODO to avoid allocating a new buffer each time we need to send a pong response.

File: peer/brontide.go
Comment (Roasbeef) [inline peer/brontide.go:82]: Hmm, sort of weird to allocate this much memory on the package level...think I'll instead make this a new required config option.

Comment (Roasbeef) [reply peer/brontide.go]: Ok re-worked to get rid of this. We use a new struct level variable, that's initialize by the server and retained across all goroutines.

File: peer/brontide.go
Comment (carlaKC) [inline peer/brontide.go:488]: `String` doesn't have `p.cfg.Conn.LocalAddr()` in it fwiw, so we're losing a little information here.

File: lnwire/ping.go
Comment (carlaKC) [inline lnwire/ping.go:50]: Why are we reading twice? can't we just return nil here?

File: peer/brontide.go
Comment (carlaKC) [inline peer/brontide.go:333]: s/accross/across

File: lnwire/pong.go
Comment (carlaKC) [inline lnwire/pong.go:16]: Linter unhappy on next commit, `ErrPongBytesExceeded`?

**Review by carlaKC — COMMENTED** [review]
Nice optimization!

**Comment by Crypt-iQ** [comment]
`TestLightningWireProtocol` is currently failing because of `Ping.Decode`. Once that is fixed, there is another possible test flake in the same test. `TestLightningWireProtocol` calls `WriteMessage` then `ReadMessage` and asserts equality. So testing.Quick may generate NumPongBytes in the range [65532, 65535], write it with no problem and then fail on ReadMessage. I think this needs an addition to the `customTypeGen` map for pings such that MaxPongBytes is respected. This doesn't mean anything bad can happen in practice as the same ping message isn't written and then read. Once other comments addressed I think this can land. Nice optimization!

File: peer/brontide.go
Comment (Roasbeef) [reply peer/brontide.go:488]: Inadvertent change during a rebase, reverted.

**Comment by Roasbeef** [comment]
@Crypt-iQ fixed the initial failure due to the double read there, but not encountering this other issue with the rest of the quick checks.

**Comment by Roasbeef** [comment]
Added a new test for that path, and also updated the quick check tests as well.

File: lnwire/lnwire_test.go
Comment (Crypt-iQ) [inline lnwire/lnwire_test.go:875]: s/int he/in the

File: lnwire/lnwire_test.go
Comment (Crypt-iQ) [inline lnwire/lnwire_test.go:877]: nit: r.Intn will generate [0, MaxMsgBody-2) when it's possible to write padding bytes of [0, MaxMsgBody-2] just fine

File: lnwire/lnwire_test.go
Comment (Crypt-iQ) [inline lnwire/lnwire_test.go:879]: nit: same here, could be [0, MaxPongBytes] for more complete coverage of possible values

**Review by Crypt-iQ — COMMENTED** [review]
lgtm after nits

**Pull request merged**
